system/lib/pthread/library_pthread.c

```diff
@@ -684,8 +683,8 @@ void emscripten_current_thread_process_queued_calls() {
     pthread_mutex_lock(&call_queue_lock);
 
     head = (head + 1) % CALL_QUEUE_SIZE;
-    emscripten_atomic_store_u32((void*)&q->call_queue_head, head);
-    tail = emscripten_atomic_load_u32((void*)&q->call_queue_tail);
+    atomic_store(&q->call_queue_head, head);
```

Comment:
What if another thread changed the current head when `call_queue_lock` was released during the `_do_call`? It seems like we should load the head again here before updating it, otherwise we might accidentally set it to a stale value.